include/fmt/format.h

```diff
@@ -465,25 +465,56 @@ class counting_iterator {
   value_type operator*() const { return {}; }
 };
 
-template <typename OutputIt> class truncating_iterator_base {
+// Used to ensure that truncating_iterator SFINAEs properly on the underlying
+// iterator's default constructibility.
```

Comment:
I think we can require default constructibility unconditionally and skip SFINAE. `truncating_iterator` is only used in format string compilation anyway.

Reply:
Done.